src/app/characters/character-list/character-list.component.ts

```diff
@@ -37,20 +40,42 @@ export class CharacterListComponent implements OnInit {
 
   trackByCharacters(_: number, character: Character) { return character.id; }
 
+  private showWarning() {
+    this.matSnackBarRef = this.snackbar.open(
+      'There are more results of your search that are not currently displayed. Please try to refine your search criteria.',
+      'Dismiss',
+      { duration: 20000 },
+    );
+  }
+
+  private dismissWarning() {
+    if (this.matSnackBarRef) {
+      this.matSnackBarRef.dismiss();
+    }
+  }
+
   private getCharacters() {
     this.characters$ = this.searchTerms.pipe(
       filter(term => term.length >= 3),
       debounceTime(300),
       distinctUntilChanged(),
       switchMap(term => {
         this.showProgress = true;
-        return this.characterService.getCharacters(term);
+        this.dismissWarning();
+        return this.characterService.getCharacters(term, 0, this.CHARACTERS_LIMIT);
       }),
-      switchMap(heroes => {
+      map(({ results: heroes, total }) => {
```

Comment:
Since we use `data` of the `MarvelResponse` type, consider creating a new interface for its `data` property and reference it directly here.

Reply:
I think we don't have to reference it here, since it is inferred automatically from the `switchMap` operator. Also, it allows us to make object deconstruction.  Do you think that if we reference an interface here, it could make the code even clearer?

Reply:
Good point 👍 and very good idea with deconstruction 💯

I was wondering: why we changed `switchMap` to `map` here? I thought that we need to return an observable to match the type of `characters$`.

Reply:
I believe that in this case, the use of the `map` operator is more appropriate because we are just transforming the data from the previous observable by returning only the list of heroes without the use of `of()` operator. Do you think that it is wrong to use it here?

Reply:
You are right 👍 I just tested it and if we return observable again, the type is `Observable<Observable<Character[]>>` which is not what we want. Thanks for the clarification 🙏